src/DappIcon/dappIcon.js

```diff
@@ -14,64 +14,81 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
-import React from 'react';
+import React, { Component } from 'react';
+import { observer } from 'mobx-react';
 import PropTypes from 'prop-types';
-
-import { createIcon } from '../Icons';
+import DappsUrlStore from '@parity/mobx/lib/dapps/DappsUrlStore';
 
 import styles from './dappIcon.css';
 
-export default function DappIcon ({ app, className, small }, { api }) {
-  const { dappsUrl } = api;
-  const classes = [
-    styles.icon, styles[small ? 'small' : 'normal'], className
-  ].join(' ');
+@observer
+class DappIcon extends Component {
+  static contextTypes = {
+    api: PropTypes.object.isRequired
+  };
 
-  if (app['semantic-icon']) {
-    return createIcon(app['semantic-icon'], { className: classes });
-  }
+  static propTypes = {
+    app: PropTypes.object.isRequired,
+    className: PropTypes.string,
+    raised: PropTypes.bool,
+    size: PropTypes.oneOf(['small', 'normal'])
+  };
 
-  let imageSrc;
-  let imageRef;
+  static defaultProps = {
+    raised: true,
+    size: 'normal'
+  };
 
-  if (app.type === 'builtin' || app.type === 'view') {
-    let dapphost = process.env.DAPPS_URL || (
-      process.env.NODE_ENV === 'production'
-        ? `${dappsUrl}/ui`
-        : ''
-    );
+  state = {
+    hasError: false // Do not show broken image when fetching image gives error
+  };
+
+  dappsUrlStore = DappsUrlStore.get(this.context.api);
+
+  handleError = () => {
+    this.setState({ hasError: true });
+  };
+
+  render () {
+    const { app, className, raised, size } = this.props;
+    const dappHost = (process.env.DAPPS_URL || `${this.dappsUrlStore.dappsUrl}/ui`).trimRight('/');
 
-    if (dapphost === '/') {
-      dapphost = '';
+    const classes = [styles.icon, raised && styles.raised, styles[size], className].join(' ');
+
+    if (!this.dappsUrlStore.dappsUrl) {
+      return <div className={classes} />; // A blank frame
     }
 
-    const fallbackSrc = window.location.protocol === 'file:'
-      ? `dapps/${app.id}/icon.png`
-      : `${dapphost}/dapps/${app.id}/icon.png`;
+    let imageSrc;
+
+    switch (app.type) {
+      case 'view': {
+        const fallbackSrc =
+          window.location.protocol === 'file:' ? `dapps/${app.id}/icon.png` : `${dappHost}/dapps/${app.id}/icon.png`;
```

Comment:
@jacogr Can you confirm that we are not using file:// anymore?

In that case I'll just put `fallbackSrc = ${dappHost}/dapps/${app.id}/icon.png`

Reply:
We are using it for Electron. (WIP, will become important going forward)

Reply:
OK so I'll leave it like this